nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/services/slack/SlackRestService.java

```diff
@@ -0,0 +1,116 @@
+                            final String apiUrl) {
+        this.webClientServiceProvider = webClientServiceProvider;
+        this.accessToken = accessToken;
+        this.apiUrl = apiUrl;
+        this.objectMapper = new ObjectMapper();
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        this.logger = LoggerFactory.getLogger(SlackRestService.class);
+    }
+
+    public void sendMessageToChannel(final String message, final String channel) throws SlackRestServiceException {
+        final URI apiUri = URI.create(apiUrl);
+        final HttpUriBuilder uriBuilder = webClientServiceProvider.getHttpUriBuilder()
+                .scheme(apiUri.getScheme())
+                .host(apiUri.getHost())
+                .encodedPath(apiUri.getPath())
+                .addPathSegment("chat.postMessage");
+        if (apiUri.getPort() != -1) {
+            uriBuilder.port(apiUri.getPort());
+        }
+        final URI uri = uriBuilder.build();
+
+        final ObjectNode requestBodyJson = objectMapper.createObjectNode();
+        if (StringUtils.isEmpty(channel)) {
+            throw new SlackRestServiceException("The channel must be specified.");
+        }
+        requestBodyJson.put("channel", channel);
+
+        if (StringUtils.isEmpty(message)) {
+            throw new SlackRestServiceException("No message to be sent with this record.");
+        }
+        requestBodyJson.put("text", message);
+
+
+        final InputStream requestBodyInputStream;
+        try {
+            requestBodyInputStream = IOUtils.toInputStream(
+                    objectMapper.writeValueAsString(requestBodyJson),
+                    StandardCharsets.UTF_8
+            );
+        } catch (final JsonProcessingException e) {
+            throw new SlackRestServiceException("JSON processing exception occurred", e);
+        }
+
+        try (final HttpResponseEntity response = webClientServiceProvider.getWebClientService()
+                .post()
+                .uri(uri)
+                .header("Authorization", String.format("Bearer %s", accessToken))
+                .header("Content-Type", "application/json")
+                .body(requestBodyInputStream, OptionalLong.of(requestBodyInputStream.available()))
+                .retrieve()) {
+            final int statusCode = response.statusCode();
+            if (!(statusCode >= 200 && statusCode < 300)) {
+                throw new SlackRestServiceException("HTTP error code: " + statusCode);
+            }
+
+            try {
+                final SlackPostMessageResponse slackResponse = objectMapper.readValue(response.body(), SlackPostMessageResponse.class);
+                logger.error(objectMapper.writeValueAsString(slackResponse));
+
+                slackResponse.checkResponse(logger);
+            } catch (final IOException e) {
+                throw new SlackRestServiceException("Slack response JSON cannot be parsed.", e);
```

Comment:
Recommend removing the period character.
```suggestion
                throw new SlackRestServiceException("JSON response parsing failed", e);
```

Reply:
Making the changes